Core/Src/compute.c

```diff
@@ -98,10 +102,46 @@ bool compute_charger_connected()
 // 	return;
 // }
 
-void compute_set_fan_speed(uint8_t new_fan_speed)
+//? Change timers to not 1 and 8 since they are advanced timers?
+uint8_t compute_set_fan_speed(uint8_t new_fan_speed, uint8_t fan_select)
 {
-	fan_speed = new_fan_speed;
-	// NERduino.setAMCDutyCycle(new_fan_speed);  Replace
+	// Define variables
+	TIM_HandleTypeDef htim;
+	TIM_OC_InitTypeDef PWMConfig;
+	uint16_t CCR_value = 0;
+
+	// Index of array +1 corresponds to which fan the channel controls
+	uint32_t channels[6] = {TIM_CHANNEL_3, TIM_CHANNEL_1, TIM_CHANNEL_4, TIM_CHANNEL_3, TIM_CHANNEL_2, TIM_CHANNEL_1};
+	
+	// Select which timer to use based on fan being initialized
+	// Based on timer parameters, determine what pulse width count would give the correct duty cycle
+	if (fan_select > 0 && fan_select <= NUM_FANS_ON_TIM1){
+		htim.Instance = TIM1;
+		CCR_value = (TIM1->ARR * new_fan_speed) / 100;
+	}
+	else if (fan_select > NUM_FANS_ON_TIM1 && fan_select <= NUM_FANS_TOTAL){
+		htim.Instance = TIM8;
+		CCR_value = (TIM8->ARR * new_fan_speed) / 100;
+	}
+	else{
+		return 1;
+	}
+
+	// Set object to how PWM channel should be configured
+	PWMConfig.OCMode = TIM_OCMODE_PWM1;
+	PWMConfig.Pulse = CCR_value;
+	PWMConfig.OCPolarity = TIM_OCPOLARITY_HIGH;
+	PWMConfig.OCFastMode = TIM_OCFAST_DISABLE;
```

Comment:
Feels a bit wasteful to instantiate this constant every time. Any chance we can keep this in a config somewhere?